### PR TITLE
ROX-16531: Test group create/update with invalid references

### PR DESCRIPTION
**Pull request by ivan-degtiarenko**
## Description

This is first PR in series of PRs to introduce e2e test for behaviour implemented in https://issues.redhat.com/browse/ROX-14700. 

This particular PR focuses on creating/updating groups with invalid role or auth provider references.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

1. Run new tests locally
2. + CI should be sufficient

### CODE REVIEW

File: qa-tests-backend/src/test/groovy/GroupsTest.groovy [hunk 1]
```diff
@@ -117,6 +120,75 @@ class GroupsTest extends BaseSpecification {
         }
     }
 
+    @Unroll
```
Comment (dhaus67) [inline qa-tests-backend/src/test/groovy/GroupsTest.groovy:123]: Does Unroll make sense if we have no custom naming pattern for the test?

File: qa-tests-backend/src/test/groovy/GroupsTest.groovy [hunk 2]
```diff
@@ -117,6 +120,75 @@ class GroupsTest extends BaseSpecification {
...
+                .setAuthProviderId(PROVIDER_IDS_BY_NAME["groups-test-provider-1"])
+                .setKey("this is so that group")
+                .setValue("will be non-default and we get invalid arguments error")
+                .build()
+        GroupService.getGroupService().createGroup(Group.newBuilder()
+                .setProps(props)
+                .setRoleName("non-existent")
+                .build())
+        then:
+        def error = thrown(StatusRuntimeException.class)
```
Comment (dhaus67) [inline qa-tests-backend/src/test/groovy/GroupsTest.groovy:136]:
```suggestion
        def error = thrown(StatusRuntimeException)
```

File: qa-tests-backend/src/test/groovy/GroupsTest.groovy [hunk 3]
```diff
@@ -117,6 +120,75 @@ class GroupsTest extends BaseSpecification {
...
+        when:
+        def props = GroupProperties.newBuilder()
+                .setAuthProviderId("non-existent-provider-id")
+                .build()
+        GroupService.getGroupService().createGroup(Group.newBuilder()
+                .setProps(props)
+                .setRoleName("Admin")
+                .build())
+        then:
+        def error = thrown(StatusRuntimeException.class)
```
Comment (dhaus67) [inline qa-tests-backend/src/test/groovy/GroupsTest.groovy:151]:
```suggestion
        def error = thrown(StatusRuntimeException)
```

File: qa-tests-backend/src/test/groovy/GroupsTest.groovy [hunk 4]
```diff
@@ -117,6 +120,75 @@ class GroupsTest extends BaseSpecification {
...
+        def updatedGroup = group.toBuilder()
+                .setRoleName("non-existent")
+                .build()
+        GroupService.getGroupService().updateGroup(
+                GroupServiceOuterClass.UpdateGroupRequest.newBuilder()
+                        .setGroup(updatedGroup)
+                        .build()
+        )
+        then:
+        def error = thrown(StatusRuntimeException.class)
```
Comment (dhaus67) [inline qa-tests-backend/src/test/groovy/GroupsTest.groovy:168]:
```suggestion
        def error = thrown(StatusRuntimeException)
```

File: qa-tests-backend/src/test/groovy/GroupsTest.groovy [hunk 5]
```diff
@@ -117,6 +120,75 @@ class GroupsTest extends BaseSpecification {
...
+                    .setAuthProviderId("non-existent")
+                    .build())
+            .build()
+        GroupService.getGroupService().updateGroup(
+                GroupServiceOuterClass.UpdateGroupRequest.newBuilder()
+                        .setGroup(updatedGroup)
+                        .build()
+        )
+        then:
+        def error = thrown(StatusRuntimeException.class)
```
Comment (dhaus67) [inline qa-tests-backend/src/test/groovy/GroupsTest.groovy:187]:
```suggestion
        def error = thrown(StatusRuntimeException)
```

File: qa-tests-backend/src/test/groovy/GroupsTest.groovy [hunk 6]
```diff
@@ -161,17 +233,11 @@ class GroupsTest extends BaseSpecification {
...
-        "groups-test-provider-2" | null  | "some-id"                                           | null  |
-                null        | ["Group3"]
-        "groups-test-provider-2" | "foo" | GROUPS_WITH_IDS["QAGroupTest-Group3"].props.getId() | "bar" |
-                "Group3"    | ["Group3"]
+        authProviderName | key | id | value | expectGroup | expectGroups
+        "groups-test-provider-1" | null  | GROUPS_WITH_IDS["QAGroupTest-Group1"].props.getId() | null  | "Group1"    | ["Group1", "Group2"]
+        null                     | "foo" | "some-id"                                           | "bar" | null        | ["Group2", "Group3"]
+        "groups-test-provider-1" | "foo" | GROUPS_WITH_IDS["QAGroupTest-Group2"].props.getId() | "bar" | "Group2"    | ["Group2"]
+        "groups-test-provider-2" | null  | "some-id"                                           | null  | null        | ["Group3"]
+        "groups-test-provider-2" | "foo" | GROUPS_WITH_IDS["QAGroupTest-Group3"].props.getId() | "bar" | "Group3"    | ["Group3"]
```
Comment (dhaus67) [inline qa-tests-backend/src/test/groovy/GroupsTest.groovy:241]: Guess we make it ugly again with the linter complaining
madge

**Review by dhaus67 — APPROVED** [review]
MInor things otherwise lgtm 🚢

File: qa-tests-backend/src/test/groovy/GroupsTest.groovy (hunk 6 above)
Comment (ivan-degtiarenko) [reply qa-tests-backend/src/test/groovy/GroupsTest.groovy:237]: I put `SupressWarnings('RuleLength')` instead. Splitting test parameter lines makes them pretty much unreadable IMHO

**Pull request merged**